modules/toxiproxy/src/main/java/org/testcontainers/containers/ToxiproxyContainer.java

```diff
@@ -0,0 +1,104 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import eu.rekawek.toxiproxy.Proxy;
+import eu.rekawek.toxiproxy.ToxiproxyClient;
+import eu.rekawek.toxiproxy.model.ToxicDirection;
+import eu.rekawek.toxiproxy.model.ToxicList;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Container for resiliency testing using <a href="https://github.com/Shopify/toxiproxy">Toxiproxy</a>.
+ */
+public class ToxiproxyContainer extends GenericContainer<ToxiproxyContainer> {
+
+    private static final String IMAGE_NAME = "shopify/toxiproxy:2.1.0";
+    private static final int TOXIPROXY_CONTROL_PORT = 8474;
+    private static final int FIRST_PROXIED_PORT = 8666;
+    private static final int LAST_PROXIED_PORT = 8666 + 31;
+
+    private ToxiproxyClient client;
+    private final Map<String, ContainerProxy> proxies = new HashMap<>();
+    private final AtomicInteger nextPort = new AtomicInteger(FIRST_PROXIED_PORT);
+
+    public ToxiproxyContainer() {
+        this(IMAGE_NAME);
+    }
+
+    public ToxiproxyContainer(String imageName) {
+        super(imageName);
+        addExposedPorts(TOXIPROXY_CONTROL_PORT);
+        setWaitStrategy(new HttpWaitStrategy().forPath("/version").forPort(TOXIPROXY_CONTROL_PORT));
+
+        // allow up to 32 ports to be proxied (arbitrary value). Here we make the ports exposed; whether or not
+        //  Toxiproxy will listen is controlled at runtime using getProxy(...)
+        for (int i = FIRST_PROXIED_PORT; i <= LAST_PROXIED_PORT; i++) {
+            addExposedPort(i);
+        }
+    }
+
+    @Override
+    protected void containerIsStarted(InspectContainerResponse containerInfo) {
+        client = new ToxiproxyClient(getContainerIpAddress(), getMappedPort(TOXIPROXY_CONTROL_PORT));
+    }
+
+    public ContainerProxy getProxy(GenericContainer container, int port) {
+        String upstream = container.getNetworkAliases().get(0) + ":" + port;
```

Comment:
Just realized.... 
This method uses `container` only for the alias (aka host).

Maybe we should have another method, `(String host, int port)`, and an overloaded method `getProxy(GenericContainer container, int port) { return getProxy(container.getNetworkAliases().get(0), port); }`?

In case someone will have some custom setup? Currently, they will have to create their own Toxy container if so

Reply:
Ah, you mean to allow people to test against a manually-named host? (which might not even be a docker container managed by Testcontainers!)

I think that's a stellar idea - it makes this module even more useful. Will add.

Reply:
Done